articles/build-rules.md

```diff
@@ -0,0 +1,89 @@
+---
+authorName: Vassilis Rizopoulos
+authorGithubUsername: damphyr
+---
+# TBD
+
+This article discusses the application of the [Separation of Concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) concept in Build & CI systems. 
+
+It is layed out as a set of rules to follow when creating build and CI infrastructure for your project. 
+
+Think of each rule as a [T.A.R.D.I.S.](https://en.wikipedia.org/wiki/TARDIS): they are bigger on the inside. 
+
+## Distinctions, distinctions
+
+To identify the separate concerns we should provide definitions for the terms: 
+
+A **build system** performs transformations in sequence according to a predetermined dependency chain to create artifacts. A subset of this is the compilation of sources to binaries.
+
+A **build management system** coordinates build system(s).
+
+### Specificity
+
+A build system is highly project specific. It is affected by toolchains, project conventions and generally can only run on a specially configured host (i.e. development environment).
+
+A build management system can run everywhere as long as it can start the build system on the appropriate host. The specificity of a build management system is limited to the number of version control systems it supports. Although theoretically it doesn't have to provide version control support it is a given that such support will be provided in the minimum feature set.
+
+### The Makefile is the build system
+
+Make, rake, Ant, MSBuild, graddle, grunt etc. **are not build systems**<sup>1</sup>. You create build systems with them. 
+
+Jenkins, TeamCity, TravisCI, BuildBot, TFS etc. are build management systems.
+
+## The Rules
+
+### 1. I am the build server
+
+Rule #1 requires that the build server follows the exact same steps as any other developer. 
+
+Expressed the other way around: Every developer has to be able to recreate locally the complete build process without deviations given the development environment and the correct version of the source tree.
+
```

Comment:
I had a lot of discussion regarding this rule. I'd even go a step further and say, that every developer has to be able to recreate the complete build process in a very easy form. We should aim for a single "build" statement (i.e. on the console, as a script...). I am missing this last statement a bit over the whole article (if you agree of course).

With regards to the article: Just stating the rule is a bit small breasted. What happens if we do not apply this rule? In my opinion, we do not adhere to the DRY (dont repeat yourself) principle. The build server knows one way to build and the developer another, and they'll differ over time.

Reply:
Having a single build statement is a bit of a trap. 

In any system of sufficient complexity (and that arbitrarily I will define as any system requiring 5+ developers to completion) having this feature actually introduces long build cycles. 

Allowing the developers multiple modular paths and yet sanctioning one authoritative way by use of the build server is a more flexible way. Ultimately most of the work is done in the build system and I am very opinionated on how a build system should be structured and what it should encompass (tl:dr; modular and **everything**)

At least three more articles on that subject ;)